File: DocumentDistiller/src/app/models/project.ts
import DocumentResult from './document-result';

export interface Project {
  id: string;
  name: string;
  description: string;
  ownerId: string;
  createdAt: string;
  lastModifiedAt: string;
  documents: DocumentResult[];
}

export interface UpdateProjectDto {
  id: string;
  name: string | null;
  description: string | null;
  documents: DocumentResult[] | null;
}

export interface CreateProjectDto {
  name: string;
  description: string;
}

File: Shared/Models/Project.cs
using System.Reflection.Metadata;

namespace Shared.Models;

public class Project
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public Guid OwnerId { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime LastModifiedAt { get; set; }
    
    public List<DocumentResult> Documents { get; set; }
}